### PR TITLE
Migrate addhost.php to lnms device:add

### PR DESCRIPTION
**Pull request by murrant**
Have snmp-scan.py call lnms device:add (make exit codes line up so this works)
Fix issue with ping only devices trying to detect os via snmp
Reorder options in device:add help and improve formatting
Add short options 1,2,3 for snmp version
Update docs to remove references to addhost.php
Fix a bit of code that was in functional code

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: addhost.php [hunk 1]
```diff
@@ -9,6 +9,8 @@
  * @copyright  (C) 2006 - 2012 Adam Armstrong
```
Comment (Jellyfrog) [inline addhost.php:9]: Imo we should just delete this file if it's features are ported to device:add now

Comment (murrant) [reply addhost.php:9]: I don't think we can... it has been around too long.

at best we can make it s stub that only prints, please use `lnms device:add`

Comment (Jellyfrog) [reply addhost.php:9]: We could symlink to artisan, and then in artisan check what script called it, and route to device:add command.

Comment (murrant) [reply addhost.php:9]: The input for addhost.php is garbage... we can't really handle it nicely no matter what we do.  I prefer this change for now, which seeks to have as few side effects as possible, but still enable legacy code removal.  (after it is removed in other places)

We can definitely do more changes later.

Comment (Jellyfrog) [reply addhost.php:9]: I meant so it just prints the help of device:add, but nvm its a bit excessive.
I'm all for replacing it with `echo "Please use lnms device:add instead";`, the project doesn't have the manpower to maintain old backward compatibilities

**Comment by laf** [comment]
Is the output supposed to replace things like `:version:` with the version used?

`  SNMP :version: No reply with community arista_eos`

File: addhost.php [hunk 2]
```diff
@@ -18,16 +20,14 @@
...
+$device = new Device;
+
 if (isset($options['g']) && $options['g'] >= 0) {
     $cmd = array_shift($argv);
     array_shift($argv);
     array_shift($argv);
     array_unshift($argv, $cmd);
-    $poller_group = $options['g'];
-} elseif (Config::get('distributed_poller') === true) {
-    $poller_group = Config::get('default_poller_group');
```
Comment (laf) [inline addhost.php:28]: Is this not needed before because it's set in `app/Console/Commands/DeviceAdd.php`?

**Review by laf — COMMENTED** [review]
Could we change:

```
Arguments:
  device spec                                        Hostname or IP to add
```

To
```
Arguments:
  hostname/ip                                        Hostname or IP to add
```

As `device spec` just seems unintuitive.

Other than that, I've tested many different ways of using `./addhost.php` and nothing seems broken.

Great work as always @murrant 👍

File: addhost.php (hunk 2 above)
Comment (murrant) [reply addhost.php:28]: Correct, but in ValidateDeviceAndCreate it sets up default values.  So there is no need to set them before hand.

**Comment by murrant** [comment]
> Is the output supposed to replace things like `:version:` with the version used?
> 
> ` SNMP :version: No reply with community arista_eos`

Fixed

**Comment by murrant** [comment]
Merging since this contains a bugfix

**Pull request merged**